recipes/protobuf/all/conanfile.py

```diff
@@ -55,22 +67,67 @@ def _cmake_install_base_path(self):
         return os.path.join("lib", "cmake", "protobuf")
 
     def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["CMAKE_INSTALL_CMAKEDIR"] = self._cmake_install_base_path.replace("\\", "/")
-        self._cmake.definitions["protobuf_BUILD_TESTS"] = False
-        self._cmake.definitions["protobuf_WITH_ZLIB"] = self.options.with_zlib
-        self._cmake.definitions["protobuf_BUILD_PROTOC_BINARIES"] = not self.options.lite
-        self._cmake.definitions["protobuf_BUILD_PROTOBUF_LITE"] = self.options.lite
-        if self.settings.compiler == "Visual Studio":
-            self._cmake.definitions["protobuf_MSVC_STATIC_RUNTIME"] = "MT" in self.settings.compiler.runtime
-        self._cmake.configure(build_folder=self._build_subfolder)
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["CMAKE_INSTALL_CMAKEDIR"] = self._cmake_install_base_path.replace("\\", "/")
+            self._cmake.definitions["protobuf_WITH_ZLIB"] = self.options.with_zlib
+            self._cmake.definitions["protobuf_BUILD_TESTS"] = False
+            self._cmake.definitions["protobuf_BUILD_PROTOC_BINARIES"] = True
+            self._cmake.definitions["protobuf_BUILD_LIBPROTOC"] = True
+            if self.settings.compiler == "Visual Studio":
+                self._cmake.definitions["protobuf_MSVC_STATIC_RUNTIME"] = "MT" in str(self.settings.compiler.runtime)
+            self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
-    def build(self):
-        for patch in self.conan_data["patches"][self.version]:
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch)
+
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "cmake", "protobuf-config.cmake.in"),
+            "@_protobuf_FIND_ZLIB@",
+            "# CONAN PATCH _protobuf_FIND_ZLIB@"
+        )
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "cmake", "protobuf-config.cmake.in"),
+            "include(\"${CMAKE_CURRENT_LIST_DIR}/protobuf-targets.cmake\")",
+            "# CONAN PATCH include(\"${CMAKE_CURRENT_LIST_DIR}/protobuf-targets.cmake\")"
+        )
+        if tools.Version(self.version) < "3.12.0":
+            tools.replace_in_file(
```

Comment:
@Croydon 
I won't object here, but why didn you use `tools.replace_in_file` here, instead of a patch?

Reply:
They would be more version specific and are much harder to maintain then those few replacements